src/remotes/auth0/index.js

```diff
@@ -62,45 +65,109 @@ function getConnectionResolvers(prefix, schemas) {
     },
   };
 }
-
+const uploader = new Uploader(process.env.UPLOADER_URL, process.env.UPLOADER_SECRET);
 export default function createAuth0Schema(domain, clientId, clientSecret) {
   const {
     findUsers,
     findUsersUncached,
     getRolesForUser,
     findUsersByRole,
-    updateUser
+    updateUser,
+    addRole
   } = query(domain, clientId, clientSecret);
 
   const resolvers = {};
+  resolvers.FileUpload = GraphQLUpload
   resolvers.Query = {
     getUser: async (_, { where, fresh }, ctx) => {
       const fn = fresh ? findUsersUncached : findUsers;
       return (await fn(where, ctx))[0] || null
     },
+    getDisplayedBadges: async (_, { where, fresh }, ctx) => {
+      const fn = findUsersUncached;
+      const user = (await fn(where, ctx))[0] || null
+      if (user.badges) {
+        const displayedBadges = user.badges.filter((b) => b.displayed === true).slice(0, 3)
+        return (displayedBadges.length > 0 ? displayedBadges : user.badges.slice(0, 3))
+      }
+      return null
+    },
     searchUsers: async (_, { where }, ctx) => findUsers(where, ctx),
     roleUsers: async (_, { roleId }, ctx) => findUsersByRole(roleId, ctx),
   };
   resolvers.Mutation = {
     updateUser: async (_, { username, updates }, ctx) => {
-      await updateUser(username, ctx, (prev) => ({
-        ...prev,
-        ...updates
-      }));
+      await updateUser(username, ctx, (prev) => {
+        const newUser = {
+          ...prev,
+          ...updates
+        }
+        newUser.name = (updates.displayNameFormat ? formatName(newUser.displayNameFormat, newUser.givenName, newUser.familyName) : newUser.name)
+        return newUser;
+      });
     },
-    grantBadge: async (_, { username, badge }, ctx) => {
-      await updateUser(username, ctx, (prev) => ({
+    updateAuthUser: async (_, { updates }, ctx) => {
```

Comment:
Could we merge this with `updateUser`. (`username` could be made optional in that method, and if it's null, the logged in user - if any - is chosen.)

The reason I would love to see them combined is because this handles a lot of validation stuff that `updateUser` doesn't right now.